### PR TITLE
Support multi-ASG pools

### PR DESCRIPTION
**Pull request by aermakov-zalando**
Support node pools with per-AZ ASGs. Corresponding kubernetes-on-aws PR: https://github.com/zalando-incubator/kubernetes-on-aws/pull/1092.

* Move subnet information to `Values`, group it by AZ
* Add a bunch of template functions (`asgSize`, `azID`, `azCount`)
* Don't suspend/resume autoscaling processes during update
* Drop unused code

### CODE REVIEW

File: provisioner/template.go [hunk 1]
```diff
@@ -184,6 +185,9 @@ func renderTemplate(context *templateContext, filePath string, data interface{})
 		"base64":                    base64Encode,
 		"manifestHash":              func(template string) (string, error) { return manifestHash(context, filePath, template, data) },
 		"autoscalingBufferSettings": autoscalingBufferSettings,
+		"asgSize":                   asgSize,
+		"azId":                      azId,
```
Comment (mikkeloscar) [inline provisioner/template.go]: nit: I would call this function `azID` (https://github.com/golang/go/wiki/CodeReviewComments#initialisms).

**Comment by mikkeloscar** [comment]
:+1:

**Comment by aermakov-zalando** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**